libraries/AP_Gripper/AP_Gripper_Servo.cpp

```diff
@@ -57,36 +49,34 @@ bool AP_Gripper_Servo::has_state_pwm(const uint16_t pwm) const
         // servo still moving....
         return false;
     }
-#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
-    if (is_releasing) {
-        gcs().send_text(MAV_SEVERITY_INFO, "Gripper load released");
-    } else {
-        gcs().send_text(MAV_SEVERITY_INFO, "Gripper load grabbed");
-    }
-#endif
     return true;
 }
 
-
 bool AP_Gripper_Servo::released() const
 {
-    return has_state_pwm(config.release_pwm);
+    return (config.state == AP_Gripper::STATE_RELEASED);
 }
 
 bool AP_Gripper_Servo::grabbed() const
 {
-    return has_state_pwm(config.grab_pwm);
+    return (config.state == AP_Gripper::STATE_GRABBED);
 }
 
 // type-specific periodic updates:
 void AP_Gripper_Servo::update_gripper() {
-#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
-    if (is_releasing && !is_released) {
-        is_released = released();
-    } else if (!is_releasing && is_released) {
-        is_released = !grabbed();
+    // Check for successful grabbed or released
+    if (config.state == AP_Gripper::STATE_GRABBING && has_state_pwm(config.grab_pwm)) {
+        config.state = AP_Gripper::STATE_GRABBED;
+    } else if (config.state == AP_Gripper::STATE_RELEASING && has_state_pwm(config.release_pwm)) {
+        config.state = AP_Gripper::STATE_RELEASED;
+    }
+
+    // re-grab if regrab_interval > 0.0
+    if (config.state == AP_Gripper::STATE_RELEASED &&
+        (is_positive(config.regrab_interval)) &&
+        (AP_HAL::millis() - action_timestamp > (config.regrab_interval * 1000.0))) {
+        grab();
```

Comment:
I think this is a different use-case to the existing re-grab behavour. This grabs if released. The current behavior in the EPM backend is to re-grab is grabbed because the magnets loose strength over time and need to be re-freshed?